### PR TITLE
feat: add AUTO_ID column type, with optional computed expression of form 'pre${mg_autoid}post'

### PR DESCRIPTION
**Pull request by connoratrug**
remaining todo:
- [ ] shorter autoid.

### CODE REVIEW

**Review by mswertz — APPROVED** [review]
I have refactored to have an AUTO_ID column. So probably you should review before merging. Looks good to me.

N.B. I noticed an unrelated bug that in schema editor I cannot add columns for a new table until I have saved the schema first.

**Review by dtroelofsprins — COMMENTED** [review]
I've tested the auto_id, having a specific dataType is a good thing I think. The computed field with the comment on how you should define the pre-post fix is nice, so you don't have to look it up first in the documentation. 
Functionality works fine, also with adding pre and post fixes. Would be nice if you could define the length of the autoID yourself as you can do in EMX1. But perhaps this is not possible as in EMX1 a sequence (with or without scrambler) is used, and uniqueness is guaranteed in this way (if you define the length according to the max number of rows you expect of course). Don't know if this also holds for these type of autoID values.

File: apps/molgenis-components/src/components/forms/RowEdit.vue
Comment (connoratrug) [inline apps/molgenis-components/src/components/forms/RowEdit.vue:139]: use const , see other const for impl exsample

File: apps/molgenis-components/src/components/forms/RowEdit.vue
Comment (connoratrug) [inline apps/molgenis-components/src/components/forms/RowEdit.vue:1]: should expect sample/demo to be updated to include new column type

File: apps/schema/src/components/ColumnEditModal.vue
Comment (connoratrug) [inline apps/schema/src/components/ColumnEditModal.vue:121]: replace with named function ( computed ) , maybe; 'isEditable(column)'

**Review by chinook25 — DISMISSED** [review]
Code looks good, just some minor code things. 
However when testing I found a bug: If I add an autoId column without computed it doesn't generate when adding or editing a row. When I edit it the row, the field shows in the form as just a string input (editing this string also gets saved).

When used with computed it seems to work fine and actually shows the ids.

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
Found a bug (?) when trying to update a row with an autoID including a computed expression:
image

And if you don't include the computed expression with pre/post fix, an update of the row works, but the ID, get "undefined"...

File: apps/molgenis-components/src/components/forms/FormInput.vue
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/FormInput.vue:173]: yes, this is what I meant with annoying formatting.

File: apps/molgenis-components/src/components/forms/RowEdit.vue
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/RowEdit.vue:1]: updated pet store

**Comment by mswertz** [comment]
> Found a bug (?) when trying to update a row with an autoID including a computed expression: image
> 
> And if you don't include the computed expression with pre/post fix, an update of the row works, but the ID, get "undefined"...

Solved

**Pull request merged**
